core/src/components/picker-column/picker-column.tsx

```diff
@@ -292,21 +297,67 @@ export class PickerColumn implements ComponentInterface {
         const centerX = bbox.x + bbox.width / 2;
         const centerY = bbox.y + bbox.height / 2;
 
-        const activeElement = el.shadowRoot!.elementFromPoint(centerX, centerY) as HTMLButtonElement | null;
+        /**
+         * elementFromPoint returns the top-most element.
+         * This means that if an ion-backdrop is overlaying the
+         * picker then the appropriate picker-column-option will
+         * not be selected. To account for this, we use elementsFromPoint
+         * and use an Array.find to find the appropriate column option
+         * at that point.
+         *
+         * Additionally, the picker column could be used in the
+         * Shadow DOM (i.e. in ion-datetime) so we need to make
+         * sure we are choosing the correct host otherwise
+         * the elements returns by elementsFromPoint will be
+         * retargeted. To account for this, we check to see
+         * if the picker column has a parent shadow root. If
+         * so, we use that shadow root when doing elementsFromPoint.
+         * Otherwise, we just use the document.
+         */
+        const rootNode = el.getRootNode();
+        const hasParentShadow = rootNode instanceof ShadowRoot;
+        const referenceNode = hasParentShadow ? (rootNode as ShadowRoot) : doc;
```

Comment:
This changed because we moved the options from inside of picker-column's shadow DOM to outside of it.